apps/aecore/test/aec_fork_signalling_tests.erl

```diff
@@ -0,0 +1,130 @@
+
+-define(FORK_CFG,
+        #{signalling_start_height => ?SIGNALLING_START_HEIGHT,
+          signalling_end_height   => ?SIGNALLING_END_HEIGHT,
+          signalling_block_count  => ?SIGNALLING_BLOCK_COUNT,
+          fork_height             => ?FORK_HEIGHT,
+          info_field              => ?INFO_FIELD_SUPPORT,
+          version                 => ?VERSION_NEW}).
+
+-define(BASIC_CHAIN_NEGATIVE_RESULT,
+        #{1                        => #{version => ?VERSION_OLD, info => ?INFO_FIELD_OTHER},
+          ?SIGNALLING_START_HEIGHT => #{version => ?VERSION_OLD, info => ?INFO_FIELD_AGAINST},
+          ?SIGNALLING_END_HEIGHT   => #{version => ?VERSION_OLD, info => ?INFO_FIELD_OTHER}}).
+
+-define(BASIC_CHAIN_POSITIVE_RESULT,
+        #{1                        => #{version => ?VERSION_OLD, info => ?INFO_FIELD_OTHER},
+          ?SIGNALLING_START_HEIGHT => #{version => ?VERSION_OLD, info => ?INFO_FIELD_SUPPORT},
+          ?SIGNALLING_END_HEIGHT   => #{version => ?VERSION_OLD, info => ?INFO_FIELD_OTHER}}).
+
+basic_chain_test_() ->
+    {foreach,
+     fun() ->
+             aec_test_utils:mock_genesis_and_forks(),
+             ?TEST_MODULE:start_link(),
+             aec_test_utils:start_chain_db(),
+             aec_test_utils:aec_keys_setup()
+     end,
+     fun(TmpDir) ->
+             aec_test_utils:aec_keys_cleanup(TmpDir),
+             aec_test_utils:stop_chain_db(),
+             ?TEST_MODULE:stop(),
+             aec_test_utils:unmock_genesis_and_forks()
+     end,
+     [{"Basic chain negative result test", fun() -> basic_chain(?BASIC_CHAIN_NEGATIVE_RESULT, false) end},
+      {"Basic chain positive result test", fun() -> basic_chain(?BASIC_CHAIN_POSITIVE_RESULT, true) end}
+     ]}.
+
+basic_chain(BlockCfgs, ExpectedForkResult) ->
+    %% FORK_HEIGHT blocks are needed to be generated (including the genesis),
+    %% so the chain ends up at H2 - 1 height where it's possible to find out
+    %% the fork signalling result.
+    Chain = [B0, B1, B2, B3, B4, B5, B6] =
+        aec_test_utils:gen_blocks_only_chain(?FORK_HEIGHT, BlockCfgs),
+    [BH0, BH1, BH2, BH3, BH4, BH5, BH6] = [block_hash(B) || B <- Chain],
+
+    ok = insert_block(B0),
+    ?TEST_MODULE:compute_fork_result(B0, BH0, ?FORK_CFG),
+    {error, not_last_block_before_fork} = ?TEST_MODULE:get_fork_result(B0, BH0, ?FORK_CFG),
+
+    ok = insert_block(B1),
+    ?TEST_MODULE:compute_fork_result(B1, BH1, ?FORK_CFG),
+    {error, not_last_block_before_fork} = ?TEST_MODULE:get_fork_result(B1, BH1, ?FORK_CFG),
+
+    ok = insert_block(B2),
+    ?TEST_MODULE:compute_fork_result(B2, BH2, ?FORK_CFG),
+    {error, not_last_block_before_fork} = ?TEST_MODULE:get_fork_result(B2, BH2, ?FORK_CFG),
+
+    ok = insert_block(B3),
+    ?TEST_MODULE:compute_fork_result(B3, BH3, ?FORK_CFG),
+    {error, not_last_block_before_fork} = ?TEST_MODULE:get_fork_result(B3, BH3, ?FORK_CFG),
+
+    ok = insert_block(B4),
+    ?TEST_MODULE:compute_fork_result(B4, BH4, ?FORK_CFG),
```

Comment:
Can two workers start concurrently? Maybe some workers can end out of order?

I am mindful that test may introduce artificial synchronization points. Would it be possible to add a test that attempts to cause multiple concurrent workers (I mean subsequent blocks) e.g. by creating large chain with large signalling interval, call `compute_fork_result` for all inserted blocks, check that outcome of computation still makes sense?